Meta/gn/create_gn_build.py

```diff
@@ -0,0 +1,149 @@
+    parser.add_argument('--third-party-prefix', type=str,
+                        help='Location of installed third-party dependencies', default='')
+    parser.add_argument('--vcpkg', type=bool,
+                        help='Whether to use vcpkg to build third-party dependencies', default=True)
+    parser.add_argument('--vcpkg-triplet', type=str, help='The vcpkg triplet to use')
+    parser.add_argument('--config', type=str, help='The build configuration to use',
+                        choices=['release', 'debug', 'sanitizer'], default='release')
+    parser.add_argument('--enable-qt', type=bool, help='Whether to enable Qt chrome',
+                        default=platform.system() != 'Darwin')
+    args = parser.parse_args()
+
+    source_dir = pathlib.Path(__file__).resolve().parent.parent.parent
+    build_dir = pathlib.Path(args.build_dir)
+    build_dir.mkdir(parents=True, exist_ok=True)
+
+    use_vcpkg = args.vcpkg
+    if args.third_party_prefix:
+        print(f'Using third-party prefix instead of vcpkg: {args.third_party_prefix}')
+        use_vcpkg = False
+    elif not args.vcpkg:
+        print('Must have either --third-party-prefix=<path> or --vcpkg=True')
+        return 1
+
+    if use_vcpkg:
+        print("-- begin vcpkg --")
+        subprocess.run([str(source_dir / 'Toolchain/BuildVcpkg.sh')], check=True)
+        vcpkg_triplet = args.vcpkg_triplet if args.vcpkg_triplet else default_vcpkg_triplet()
+        vcpkg_install_dir = build_dir / 'vcpkg_installed'
+        vcpkg_root = source_dir / 'Toolchain/Tarballs/vcpkg'
+        vcpkg_binary = vcpkg_root / 'vcpkg'
+        overlay_triplets = source_dir / f'Meta/CMake/vcpkg/{args.config}-triplets'
+        binary_cache = source_dir / 'Toolchain/Build/vcpkg-binary-cache'
+
+        env = os.environ.copy()
+        env['CC'] = args.cc
+        env['CXX'] = args.cxx
+        subprocess.run([vcpkg_binary, 'install', f'--triplet={vcpkg_triplet}', f'--vcpkg-root={vcpkg_root}',
+                        '--x-wait-for-lock', f'--x-manifest-root={source_dir}', '--no-print-usage',
+                        f'--overlay-triplets={overlay_triplets}',
+                        f'--binarysource=clear;files,{binary_cache},readwrite',
+                        f'--x-install-root={vcpkg_install_dir}'],
+                       env=env, check=True)
+        print("-- end vcpkg --")
+
+        third_party_prefix = vcpkg_install_dir.resolve() / vcpkg_triplet
+    else:
+        third_party_prefix = pathlib.Path(args.third_party_prefix).resolve()
+
+    # FIXME: Should we grab qt from vcpkg or use the system qt?
+    enable_qt = args.enable_qt
+
+    is_clang = False
+    cc = pathlib.Path(args.cc)
+    cxx = pathlib.Path(args.cxx)
+    if cc.name.startswith('clang') or cxx.name.startswith('clang'):
+        is_clang = True
+
+    args_gn_contents = f'''
+host_cc="{cc}"
+host_cxx="{cxx}"
+is_clang={gn_bool(is_clang)}
+use_lld={gn_bool(platform.system() != 'Darwin')}
```

Comment:
Two things:

1. I wouldn't clobber args.gn if it exists but write some gni file that args.gn imports, so that people can tweak their args.gni
2. I wouldn't write anything in here that could live in gn files themselves (like the use_lld thing; sanitizer or debug settings, host cc, etc – basically nothing but third_party_prefix and qt stuff)

Reply:
I'm not sure how to do the "write some gni file that args.gn imports" bit, but I did replace this script with Meta/gn/build/libs/BUILD.gn and a bunch of vcpkg variables in Meta/gn/build/vcpkg.gni.

Meta/gn/build/libs/third_party.gni now contains the flags to disable vcpkg and set your own third_party_prefix if desired.

Reply:
I guess it should be said, I'll need to clean up this branch before actually merging it. Possibly by using compnerd's organization for the first set of commits, though there's some erroneous changes in there that I'll have to fix to get a proper working build.